src/librustdoc/html/render/cache.rs

```diff
@@ -373,18 +373,14 @@ crate fn get_all_types<'tcx>(
     tcx: TyCtxt<'tcx>,
 ) -> (Vec<TypeWithKind>, Vec<TypeWithKind>) {
     let mut all_types = Vec::new();
+    let mut args = Vec::new();
     for arg in decl.inputs.values.iter() {
         if arg.type_.is_self_type() {
             continue;
         }
-        // FIXME: performance wise, it'd be much better to move `args` declaration outside of the
-        // loop and replace this line with `args.clear()`.
-        let mut args = Vec::new();
         get_real_types(generics, &arg.type_, tcx, 0, &mut args);
         if !args.is_empty() {
-            // FIXME: once back to performance improvements, replace this line with:
-            // `all_types.extend(args.drain(..));`.
-            all_types.extend(args);
+            all_types.append(&mut args);
```

Comment:
Why not just move this line out of the loop?

IIUC, `args` is a transformed version of `decl.inputs`, so shouldn't we just be appending to `args` inside the loop and then extending `all_types` with it at the end? This change would both improve code clarity and potentially improve performance.

Reply:
It's because we check if something was added into `args` and do something special in case it was empty. At least that would be my guess from my understanding of the code.

Reply:
I didn't see any checks of the contents of args in the functions called from inside the loop. But I could have missed something.

Reply:
We discussed this and realized that the check happens directly above this line 😆